### DISCUSSION TITLE
ESP8266 Over ESPHome to insert Temps

### DISCUSSION THREAD

**Discussion by Sammerdan** [freakent/dbus-mqtt-devices#139] [Q&A]
Hi all,

I've been using Home Assistant now for a while. Since I have a lot of ESP devices in my Van (I love temps). I was thinking, This could be better and direct into  my Venus. So I found this dbus-mqtt! GREAT!

So, I havn't got a lot of programming experience, but got ESPHome working on Home Assistant.

I like the overview of my devices in my Dashboard and the availability to OTA update them.

Now I wanted to test with 1 of them to communicate with my Cerbo.
I followed your guide and let ChatGPT help me to set u a YAML, but it doesn't work.

It gets stuck in step 3 (No device/tanksensoren/DBus)

What am I doing wrong?

Here My YAML
```
esphome:
  name: tank-sensoren
  friendly_name: Tank_Sensoren

esp8266:
  board: esp01_1m

# Enable logging
logger:

# Enable OTA updates
ota:
  - platform: esphome
    password: "secrte"

wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_password

  # Enable fallback hotspot (captive portal) in case wifi connection fails
  ap:
    ssid: "Tank-Sensoren Fallback Hotspot"
    password: "secrte"

mqtt:
  broker: 192.168.1.126
  topic_prefix: "device/tanksensoren"
  discovery: true

  # Birth and Will messages
  birth_message:
    topic: "device/tanksensoren/Status"
    payload: |-
      {
        "clientId": "tanksensoren",
        "connected": 1,
        "version": "v1.0",
        "services": {
          "t1": "temperature",
          "t2": "temperature",
          "t3": "temperature",
          "h1": "humidity"
        }
      }

  will_message:
    topic: "device/tanksensoren/Status"
    payload: |-
      {
        "clientId": "tanksensoren",
        "connected": 0
      }

# Handle the DBus topic subscription by using on_message
  on_message:
    topic: "device/tanksensoren/DBus"
    then:
      - logger.log: "Received DBus data"
      - logger.log: "{{ payload }}"


# One-wire Bus Configuration for temperature sensors
one_wire:
  platform: gpio
  pin: GPIO13

sensor:
  - platform: dallas_temp
    address: "0xB00120420D898028"
    name: "WaterTanktempLinks"
    id: t1
    on_value:
      then:
        - mqtt.publish:
            topic: "W/"POrtalID"/temperature/5/Temperature"
            payload: !lambda |-
              char buffer[40];
              snprintf(buffer, sizeof(buffer), "{\"value\": %f}", id(t1).state);
              return std::string(buffer);

  - platform: dallas_temp
    address: "0xC10120424FB79528"
    name: "WaterTanktempRechts"
    id: t2
    on_value:
      then:
        - mqtt.publish:
            topic: "W/"POrtalID"/temperature/6/Temperature"
            payload: !lambda |-
              char buffer[40];
              snprintf(buffer, sizeof(buffer), "{\"value\": %f}", id(t2).state);
              return std::string(buffer);

  - platform: dht
    pin: GPIO5
    model: DHT22
    temperature:
      name: "Temperatuurtussenbodem"
      id: t3
      on_value:
        then:
          - mqtt.publish:
              topic: "W/"POrtalID"/temperature/7/Temperature"
              payload: !lambda |-
                char buffer[40];
                snprintf(buffer, sizeof(buffer), "{\"value\": %f}", id(t3).state);
                return std::string(buffer);

    humidity:
      name: "Vochtigheidtussenbodem"
      id: h1
      on_value:
        then:
          - mqtt.publish:
              topic: "W/"POrtalID"/humidity/8/Humidity"
              payload: !lambda |-
                char buffer[40];
                snprintf(buffer, sizeof(buffer), "{\"value\": %f}", id(h1).state);
                return std::string(buffer);

captive_portal:
```
Offcourse the "POrtalID" is set to mine.

**Comment by freakent** [comment]
Hi,

I took the liberty of formatting your question so the YAML is readable. 

I'm not actually familiar with this style of programming but I can see a couple of things that do not look correct. 

1) you have a h1 as being a device  type of humidity. This is not correct. In Victron's dbus humidity is not a separate sensor, it is an attribute of temperature. So you can send /Temperature, /Pressure and /Humidity to a temperature type device. Check the services.yaml file to see the attributes defined for each device type (or check out Victron's dbus api wiki page). 

2. You must use the device id returned in the Dbus message in the topic path for each published reading. The whole reason for device registration is to obtain a device id from the dbus. You appear to be hard coding 5, 6, 7, 8 in the topic path instead of using the value returned in registration. The topic path including the device id is now also returned in the Dbus message to make life easier for you.

**Comment by Sammerdan** [comment]
Hi, Thank you!

Yeah, I tried the hard way. Now I found that it's way easier to use the
Node-RED solution.

Thanks anyway!

Op di 26 nov 2024 om 23:56 schreef Martin Jarvis ***@***.***>:
